api/scopes/views.py

```diff
@@ -30,14 +31,14 @@ class ScopeDetail(JSONAPIBaseView, generics.RetrieveAPIView):
     # overrides RetrieveAPIView
     def get_object(self):
         id = self.kwargs[self.lookup_url_kwarg]
-        scope_item = public_scopes.get(id, None)
-        if scope_item:
-            scope = Scope(id=id, scope=scope_item)
-            self.check_object_permissions(self.request, scope)
-            return scope
-        else:
+        try:
+            scope = ApiOAuth2Scope.objects.get(name=id)
```

Comment:
I see one major change is that we are replacing `api.scopes.serializers.Scope` with `osf.models.oauth.ApiOAuth2Scope`.

* Correct me if wrong that: I think this is also a favorable improvement 👍  **independent of** `scopes` being relationships now.
* In addition, we can remove the class `api.scopes.serializers.Scope` since it is no longer used anywhere in the code base.